Simplenote/src/main/java/com/automattic/simplenote/utils/IntentUtils.java

```diff
@@ -0,0 +1,27 @@
+package com.automattic.simplenote.utils;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+
+public class IntentUtils {
+    public static String getMainActivityClassName(Context context) {
+        String packageName = context.getPackageName();
+        Intent launchIntent = context.getPackageManager().getLaunchIntentForPackage(packageName);
+        if (launchIntent != null && launchIntent.getComponent() != null) {
+            ComponentName componentName = launchIntent.getComponent();
+            return componentName.getClassName();
+        }
+
+        return "com.simplenote.automattic.NotesActivity";
+    }
+
+    public static Intent maybeAliasedIntent(Context context) {
+        String packageName = context.getPackageName();
+
+        Intent intent = new Intent();
+        intent.setComponent(new ComponentName(packageName, getMainActivityClassName(context)));
+
+        return intent;
```

Comment:
I added this function to get the proper main activity, if it is the alias or not. This fixes the back button crash and probably some other scenarios where we were referring directly to the `NotesActivity` class which may or may not exist depending on if the activity alias for the sustainer icon is active.